docs/DevelopmentPlan/DevelopmentPlan.tex

```diff
@@ -167,16 +202,93 @@ \section{Expected Technology}
+website should be easily maintainable. Implementing the website using an 
+extremely popular and widespread framework such as React means that there 
+are countless resources online to help future maintainers keep the project alive.
+
 \item Specific libraries
+\subitem Depending on the implementation, the middleware will use some combination
+of the FastAPI and SQLalchemy libraries. FastAPI will be used to implement HTTPS 
+commmunication rotuers between the database and webserver. SQLalchemy will be used to 
+implement the database if it is decided that it will be implemented using an ORM.
+\subitem The webserver will use a large set of both funcational and visual react
+libraries. For example react-navigation will be used for its page traversal 
+functionality while libraries such as react-datepicker and react-calendar will 
+be very helpful in implement the visual elements of a scheduling system. 
+Additionally, the axios library will be used to form and send the HTTPS requests 
+to the middleware from the webserver.
 \item Pre-trained models
+\subitem This project does not include an AI component and will not use 
+a machine learning model.
 \item Specific linter tool (if appropriate)
+\subitem ESLint will be used for linting Javascript code while flake8 will be 
+used to lint Python code. We chose these linters due to our previous experience 
+using them and because they provide our preferred formatting style.
 \item Specific unit testing framework
+\subitem The pytest framework will be used to create unit tests for the middleware
+code. We chose pytest over other python testing frameworks due to its simplicity, 
+small amount of boilerplate code, and plugins which can add useful functionalities
+like coverage reporting. We plan to incorporate these pytest unit tests as a part
+of our CI plans for the project via Github actions.
+\subitem Testing the database will likely be done using a dummy / development 
+PostgreSQL database prior to making any changes to the production database to 
+ensure that minimal migrations are required during development.
+\subitem Our plans for testing our webserver's React code are not decided yet, 
+but we're currently investigating potential testing options.
 \item Investigation of code coverage measuring tools
+\subitem The Coverage.py Python library will be used to measure the code coverage
+of our middleware program. For the webserver's React code, Jest is included by 
+default when using the "create-react-app" command and will be used to measure 
+the code coverage of the webserver.
 \item Specific plans for Continuous Integration (CI), or an explanation that CI
   is not being done
 \item Specific performance measuring tools (like Valgrind), if
   appropriate
+\subitem lighthouse for react, cProfile included with python
+\subitem The webserver will be hosted on azure's web app services, meaning that
+azure's suite of performance measrement tools and metrics will be used as the 
```

Comment:
measrement --> measurement
also above, make sure to capitalize the "L" in "lighthouse"